oneandone/resource_oneandone_server.go

```diff
@@ -555,7 +543,18 @@ func resourceOneandOneServerDelete(d *schema.ResourceData, meta interface{}) err
 
 	_, ok := d.GetOk("ip")
 
-	server, err := config.API.DeleteServer(d.Id(), ok)
+	server, err := config.API.GetServer(d.Id())
+	if err != nil {
+		return err
+	}
+
+	err = config.API.WaitForState(server, "POWERED_ON", 50, config.Retries)
```

Comment:
Shouldn't the server be POWERED_ON by the time it gets to delete?

Reply:
It is there as a safeguard. When removed, the behavior is inconsistent, which you can see from the following gist:
https://gist.github.com/aajdinov/ceecf70ccae77be0684b317aa4406318

When present, the acceptance tests pass consistently.